src/java_bytecode/java_types.h

```diff
@@ -76,4 +78,124 @@ bool is_java_array_tag(const irep_idt& tag);
 
 bool is_valid_java_array(const struct_typet &type);
 
+
+/// class to hold a Java generic type
+/// upper bound can specify type requirements
+class java_generic_typet:public reference_typet
+{
+public:
+  typet bound;
```

Comment:
Is it OK to add fields to a irep class? I would worry that sometimes they are passed around by value - though I don't know, if there are other irep's that do this I guess it is fine?

Reply:
Specifically, we return `typet` by value from `java_type_from_string` in `java_types.cpp`, won't these extra fields be sliced away?